gecko/core/container.py

```
@@ -273,22 +342,113 @@ def _create_instance(self, descriptor: ServiceDescriptor[T]) -> T:
+        # 循环依赖检测
+        if service_type in self._resolving:
+            raise CircularDependencyError(f"Circular dependency detected while resolving {service_type.__name__}")
+
+        try:
+            self._resolving.add(service_type)
+            return await self._create_instance_async(descriptor)
+        finally:
+            self._resolving.discard(service_type)
+
+    async def _create_instance_async(self, descriptor: ServiceDescriptor[T]) -> T:
+        """异步创建实例的实现，支持 async 工厂与 async 关闭"""
```

Ruff (0.14.7)

Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)